packages/package-utils/ExtendedNonceManager.js

```diff
@@ -22,25 +38,23 @@ class ExtendedNonceManager extends NonceManager {
   }
 
   async sendTransaction(transactionRequest) {
-    // What nonce are we going to attach to this?
-    // Definitely not any we've sent and are pending
-    const pendingNonces = Object.keys(this.signedTransactions).map((txhash) => this.signedTransactions[txhash].nonce);
-
-    // At least whatever the endpoint says, or whatever we've already sent if higher
-    let nonce = await this.signer.getTransactionCount();
-
-    // Note the order we did the above two lines in - if a tx is mined between these two lines,
-    // and got removed by the `on block` handler above, by doing it in this order we won't be tripped up
-    // And we'll skip any nonces we've already used
-    while (pendingNonces.includes(nonce)) {
-      nonce += 1;
+    try {
```

Comment:
This is a pretty different implementation. One big difference is that we're no longer setting nonces unless there is a failure -- are we assuming that the `transactionRequest` comes with a nonce already set?